fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/populator/office/OfficeWorkbookPopulator.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    private void setLookupTable(Sheet officeSheet) {
+        int rowIndex=1;
+        for (OfficeData office:offices) {
+            Row row=officeSheet.createRow(rowIndex);
+            writeString(LOOKUP_OFFICE_COL,row,office.name());
+            writeLong(LOOKUP_OFFICE_ID_COL,row,office.getId());
+            rowIndex++;
+        }
+    }
+
+    private void setLayout(Sheet worksheet){
+        Row rowHeader=worksheet.createRow(0);
+        worksheet.setColumnWidth(OFFICE_NAME_COL, 4000);
+        worksheet.setColumnWidth(PARENT_OFFICE_NAME_COL,4000);
+        worksheet.setColumnWidth(PARENT_OFFICE_ID_COL,4000);
+        worksheet.setColumnWidth(OPENED_ON_COL,3000);
+        worksheet.setColumnWidth(EXTERNAL_ID_COL,3000);
+        worksheet.setColumnWidth(LOOKUP_OFFICE_COL,4000);
+        worksheet.setColumnWidth(LOOKUP_OFFICE_ID_COL,4000);
+
+        writeString(OFFICE_NAME_COL, rowHeader, "Office Name*");
+        writeString(PARENT_OFFICE_NAME_COL, rowHeader, "Parent Office*");
+        writeString(PARENT_OFFICE_ID_COL,rowHeader,"Parent OfficeId*");
+        writeString(OPENED_ON_COL, rowHeader, "Opened On Date*");
+        writeString(EXTERNAL_ID_COL, rowHeader, "External Id*");
+        writeString(LOOKUP_OFFICE_COL, rowHeader, "Lookup Offices");
+        writeString(LOOKUP_OFFICE_ID_COL,rowHeader, "Lookup OfficeId*");
+    }
+
+    private void setRules(Sheet workSheet){
+        CellRangeAddressList parentOfficeNameRange = new  CellRangeAddressList(1, SpreadsheetVersion.EXCEL97.getLastRowIndex(), PARENT_OFFICE_NAME_COL, PARENT_OFFICE_NAME_COL);
+        CellRangeAddressList OpenedOndateRange = new CellRangeAddressList(1, SpreadsheetVersion.EXCEL97.getLastRowIndex(),OPENED_ON_COL,OPENED_ON_COL);
+
+        DataValidationHelper validationHelper=new HSSFDataValidationHelper((HSSFSheet) workSheet);
+        setNames(workSheet);
+
+        DataValidationConstraint parentOfficeNameConstraint=validationHelper.createFormulaListConstraint("Office");
+        DataValidationConstraint openDateConstraint=validationHelper.createDateConstraint(DataValidationConstraint.OperatorType.LESS_OR_EQUAL,"=TODAY()",null,"dd/mm/yy");
+
+        DataValidation parentOfficeValidation=validationHelper.createValidation(parentOfficeNameConstraint,parentOfficeNameRange);
+        DataValidation openDateValidation=validationHelper.createValidation(openDateConstraint,OpenedOndateRange);
+
+        workSheet.addValidationData(parentOfficeValidation);
+        workSheet.addValidationData(openDateValidation);
+    }
+
+    private void setNames(Sheet workSheet) {
+        Workbook officeWorkbook=workSheet.getWorkbook();
+        Name parentOffice=officeWorkbook.createName();
+        parentOffice.setNameName("Office");
+        parentOffice.setRefersToFormula("Offices!$H$2:$H$"+(offices.size()+1));
+    }
+    private void setDefaults(Sheet worksheet) {
+        try {
+            for (Integer rowNo = 1; rowNo < 3000; rowNo++) {
```

Comment:
don't use directly any value like 3000, define it as static variable and then use that . Do the same changes at other places also.